sklearn/decomposition/_nmf.py

```diff
@@ -1335,7 +1345,7 @@ def transform(self, X):
             beta_loss=self.beta_loss, tol=self.tol, max_iter=self.max_iter,
             alpha=self.alpha, l1_ratio=self.l1_ratio, regularization='both',
             random_state=self.random_state, verbose=self.verbose,
-            shuffle=self.shuffle)
+            shuffle=self.shuffle, callbacks=getattr(self, '_callbacks', []))
```

Comment:
I feel like we should either decide that
- no callbacks means an empty list
- no callabacks means None and having callbacks means a non-empty list

but it seems that the code is mixing both right now?

If we ultimately plan on having `callbacks=None` as a default param then the latter would be more appropriate?

Reply:
Actually, here we would still need to do provide the default option to `getattrs`:  `getattr(self, '_callbacks', None)` and that's not really much more readable than `getattr(self, '_callbacks', [])` so both would work..